src/tdywy.c

```diff
@@ -916,7 +916,7 @@ PetscErrorCode TDyWYResidual(TS ts,PetscReal t,Vec U,Vec U_t,Vec R,void *ctx) {
   PetscFunctionBegin;
   PetscErrorCode ierr;
   TDy      tdy = (TDy)ctx;
-  DM       dm;
+  DM       dm = tdy->dm;
```

Comment:
Is this value actually needed? It looks like it's set in line 925. In cases like this, one thing you can do is just declare the DM in line 924, to reflect that you're not using it until it's set.